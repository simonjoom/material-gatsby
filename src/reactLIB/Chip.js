import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const Chip = ({ children, close, className, ...props }) => {
  return (
    <div className={cx('chip', className)} {...props}>
      {children}
      {close ? <i className="close material-icons">close</i> : null}
    </div>
  );
};

Chip.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node,
  /**
   * Shows a close icon
   */
  close: PropTypes.bool
};

Chip.defaultProps = {
  close: false
};

export default Chip;
