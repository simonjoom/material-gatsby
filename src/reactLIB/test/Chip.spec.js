import React from 'react';
import { shallow } from 'enzyme';
import Chip from '../src/Chip';

describe('<Chip />', () => {
  let wrapper;

  test('should render', () => {
    wrapper = shallow(<Chip />);
    expect(wrapper).toMatchSnapshot();
  });

  test('accepts a close prop', () => {
    wrapper = shallow(<Chip close />);
    expect(wrapper.find('i.material-icons.close')).toHaveLength(1);
  });

  test('accepts children', () => {
    let child = <p className="child">Child</p>;
    wrapper = shallow(<Chip close>{child}</Chip>);
    expect(wrapper).toMatchSnapshot();
  });

  test('accepts className', () => {
    wrapper = shallow(<Chip className="find" />);
    expect(wrapper.hasClass('find')).toBeTruthy();
  });
});
