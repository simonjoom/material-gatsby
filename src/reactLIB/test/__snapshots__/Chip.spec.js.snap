// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chip /> accepts children 1`] = `
<div
  className="chip"
>
  <p
    className="child"
  >
    Child
  </p>
  <i
    className="close material-icons"
  >
    close
  </i>
</div>
`;

exports[`<Chip /> should render 1`] = `
<div
  className="chip"
/>
`;
